driver-core/src/main/com/mongodb/internal/connection/MongoCredentialWithCache.java

```diff
@@ -63,15 +67,34 @@ public void putInCache(final Object key, final Object value) {
         cache.set(key, value);
     }
 
+    public OidcCacheEntry getOidcCacheEntry() {
+        return cache.oidcCacheEntry;
+    }
+
+    public void setOidcCacheEntry(final OidcCacheEntry oidcCacheEntry) {
+        this.cache.oidcCacheEntry = oidcCacheEntry;
+    }
+
+    public <V> V withOidcLock(final Supplier<V> k) {
+        return Locks.withLock(cache.oidcLock, k);
+    }
```

Comment:
This lock does not belong in `MongoCredentialWithCache`/`MongoCredentialWithCache.Cache` as it is not even used here. It should instead be part of `OidcAuthenticator` internals.

If it were used both in `MongoCredentialWithCache`/`MongoCredentialWithCache.Cache` and in `OidcAuthenticator`, that would have been a strong indicator that class boundaries are set incorrectly.

Reply:
The lock must be respected by different connections, which each have their own OidcAuthenticator. The Cache specifically is what is shared between them.